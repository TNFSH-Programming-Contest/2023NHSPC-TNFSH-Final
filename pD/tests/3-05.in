2000 1
156350227
705144065
-161339874
793545780
231606328
452864621
553527394
637609647
264602162
199053927
-4675754
826873810
-275471217
106043464
699901471
-75848919
648076943
428077144
62444821
671746345
434153611
559746577
196517439
690295988
615342527
566877001
-104850168
-129099286
248740993
178089113
957218322
-411718179
-175446245
-71466896
-425479413
164674077
850869047
236740436
201980800
-186967695
-379663590
223814730
718419068
514267985
589528804
502111822
254335605
-410964599
432750938
515855091
-419445110
2568234
-265966096
-484748430
-14141682
576525706
958585824
17203387
-125890522
497878437
503924275
717449023
287520020
238862352
53755636
979393300
495772421
-62833291
926569081
-483025220
563742654
-467368946
-238327777
85595440
-457967282
-236601633
192608417
148050631
549680237
525061399
177410460
-312691322
264638649
-243508633
392153762
-314876182
157627252
678976788
-163043917
678849658
58784576
656712180
-249165723
163772225
785877840
-54613143
196897720
-76345959
-81029258
263091030
-415262464
-97656790
106541198
264870896
184672406
-246143641
503054510
363634680
-58006986
-415879977
170424055
587363708
667321053
160123677
425521369
-44117259
937555086
888902611
769941749
-117635464
599800690
379709877
630083161
78288373
923835977
822021224
-66779971
-346967188
599741583
722442847
-246596193
824824881
676261937
-455472165
997808270
-447526956
-260584130
663984818
-206541567
60100883
609653933
273422013
-20856522
-38257878
474633473
837785050
56919673
-320378700
783363625
-401475859
146306145
560260711
-193395447
572595209
417778592
-155747653
582348804
-313780265
-364280365
-427561703
812282947
-239570359
-477898026
585458410
-209559472
-236195772
701777395
139022408
224197357
826883415
676473798
72178365
261874314
187934336
33271828
350195695
842120459
-226940906
743696693
839400600
484481622
-415889634
318741384
611675336
-453675506
-267757370
-371925167
220580776
609468344
256420095
-150648525
757452765
726100310
52618831
-118213776
-164829661
943670717
427420672
269834611
228198816
-252357934
-253090910
780019829
191820568
-308710403
360086021
988431689
-459234268
404569956
728633863
-221718606
-137866431
424528343
-210228077
-162590621
970586119
-496382929
528522687
484262097
599442737
157120530
203629016
-1176567
623733471
960253847
605197363
431441953
-264645551
466360717
-248573625
-178724177
45199242
-468334052
-263920135
599083023
709222146
418437538
-140922209
603896042
65006913
360915135
-41338903
978574943
-150085567
990172189
-56482668
778928054
578991825
483619880
431380363
603603070
-489429707
-92616643
387251423
-229541927
-90672909
999048247
531978005
973179852
-398594803
-475049815
524759290
-316491740
-196098814
746355763
-491844772
134031905
-373354090
171226322
569754426
881501558
564868612
12858992
64778156
829947766
289658050
-37310279
18434219
812598099
549904717
505413952
650118732
-277822493
502076370
-189462668
22087244
-224650043
-307976472
-417512107
-119803836
612877542
801197855
763767432
-108704518
-244361321
741878740
334365041
675611491
436329766
240868671
913719153
116527064
951455335
-317566708
768679727
-76703715
-362693473
274933431
-341096633
381791839
270589613
-40707868
509573672
-325188607
16885113
829623468
492033535
174339579
-41412008
378828925
338655011
50151341
-217386662
360075985
507333435
-136480445
-174355457
85494937
736909134
-76230944
180835742
114438988
-252787691
671965375
391531570
307143433
325844862
-131244158
628726295
-258819184
-318763840
42130292
217896728
976502926
-438657392
514281283
827390684
544345280
-450069568
769349574
462889181
692771244
792162410
879263167
80563415
448851963
-418325981
925119054
-240522253
612152390
-488942571
127783121
3585400
-257327100
-266427174
721457944
-181674787
-29903803
290328953
351725014
-98757826
577003362
438995057
536790166
45616748
751060853
-262361603
-472904293
212879509
-231777866
258900262
-348038575
-366859751
282856095
999431084
383914722
-177673027
13543782
-252765682
139145745
903614104
499873303
-394358740
-467844368
371130143
859072296
606559672
-21745951
52985617
-349343277
626960425
450355804
177944509
764893344
808076880
-177008725
514498614
571982412
925547848
751550785
649253189
-189934408
97447886
508050548
-212802641
-202092544
587223716
291110476
967858533
239454658
-5386791
-460635783
686293513
-314239966
604360365
-258916666
124593122
799950873
518428194
903898035
663385745
867554618
456537424
879222746
218719268
399689809
-108853779
318825314
-148050088
14486283
-421452899
32047967
287189873
31242683
543922793
947981068
514152409
85215459
383833465
-221751438
379896848
268773385
-201977190
-258687390
-184549044
623569485
623830434
817094430
-63127801
549108044
516498795
674012357
-261259162
-323917458
562714829
732317751
110737653
-481299280
528381387
-475054950
-272085163
837738283
-65050142
783893229
971541948
524360518
371470674
-122607289
154721726
25121578
-71273930
58005241
-43655883
12847752
-291938752
-460073463
556926385
-154036260
308321485
681876249
383930421
-486237062
-42742406
390974078
-267041977
-74967897
-315436183
87208101
29875548
703053668
85715283
181069304
-493827376
-162163864
-108703308
-314312736
859548821
147315164
-6870126
-303829363
-402558663
827003471
459739126
-315879154
-410000266
227452214
-310277954
224389903
578042059
-296865965
560309183
-26000447
964699004
709904160
372383257
805471899
558971595
843412815
150929958
-460729271
-60802502
-248130245
-238208727
129412910
926490851
-432243707
249694672
58808583
258294403
-470490447
619420766
719973592
-271492562
738045922
864166238
631727969
691226075
-264459081
128131756
882034154
15395041
533108000
475826853
-191758078
287051720
549768329
263193266
351384515
311609812
-200540977
166418720
828942709
-21324356
-56295355
758206495
-447104585
-297539156
-104214387
62315284
-328826028
-434049827
-332586988
217297695
-171537534
197062485
572321376
456367306
812726390
-177004860
-203759939
170287206
-431628836
-256641860
-276596105
453705579
-394750924
615770861
992138599
736060296
-274166632
-247783243
65147014
-497345319
240250151
919447687
322921115
-179608181
-448993792
-53779826
-212660217
-390655502
-313122670
608818449
-193017301
782422269
-317790043
-8321677
-303232898
46021850
564105641
70091326
856492301
-185485383
961671543
343706936
447351533
957270148
992362542
416188943
147882988
580281246
-140295835
376399284
-129416569
851937547
340257377
876566094
74157182
59661363
-363727564
111996633
799866573
308665948
172911801
207561279
399375690
-341732145
896506864
429625418
961271852
75063423
-69775749
856098276
95362009
166000131
620861062
367079205
-46128278
-316206082
-362991167
134421940
879511624
941030100
355366934
230324234
-163981394
-224067839
312410548
120923377
-300186347
-454337866
607162470
-219016725
544119224
603462208
186999228
-83093341
655837125
154929853
756200707
357980861
888443104
-494350446
590917529
720596027
446726075
-63271461
-475413360
613255847
75671976
351772512
323668429
-494569943
-362674481
-14081079
-15567889
235896199
740326981
645531252
533461098
-236890100
683276435
-192949908
811458127
-76229394
476098022
-100867464
647778500
536994091
60313411
135134503
659719998
21586377
937854332
-311464971
350880551
433186194
929487149
56195883
-143652652
151707946
836257342
604302787
196916163
-340405100
-376669334
249589006
240154516
212501654
396119317
-432718644
-340730306
648699693
872679402
-394119996
457286532
9908725
381416193
512765764
534833673
531707338
836476817
-159847104
-406999502
972088485
16357283
-96323563
984415408
-374957234
612895860
316994532
-101257551
123313366
664401434
99554692
-185530716
-456192669
857168845
833666293
784448159
-261105216
821247014
322002947
87874169
-218636211
557919998
-448691407
52633653
918003638
-35720197
635635774
-34842357
916690989
938903007
295526410
609389139
320384175
85552567
344946190
828367780
49412052
396022781
-312340964
-151677782
236982904
-316074114
701715081
851034535
205842775
-222668304
454314187
961925321
318102722
2544057
20169973
135642270
537872669
644833115
-100437866
-399136165
74170788
213027667
376336833
-338258591
-494783495
951596716
906370263
-55753858
461235628
34739736
-17889484
-301313244
647149805
-445215637
114199468
-355349273
458349797
744574349
656009454
-103589352
604078819
232015140
-323949148
221333619
854823702
361056011
-71668681
999925387
-315956374
825869138
325244685
-491504254
569235122
-328120126
513690101
-121768444
-379114107
963656961
186129229
-30728801
-486545010
-212026342
903210797
128540284
933863821
-161353727
768590426
411799319
911905007
97108024
-38414021
68158217
-497587636
395079582
69843880
987262400
703991558
862318952
-81784981
639001390
176778346
-339394618
447774065
-98434329
703522165
857218972
-254674133
340284947
253148954
-233502130
230569972
220070458
565126352
996731228
316388999
632240387
92685633
235883394
345748099
312987131
776790480
835487704
342747153
448721222
423966742
-393245029
-312160063
158313021
-84093398
-121136716
291907085
-148102489
439926333
452699463
-397724120
444711494
171728584
115545210
942661095
874016291
61246382
-364976967
-112849606
-216359391
202482490
29798577
690964173
359637132
-113837561
800893617
-90176203
-453042086
-103133419
777416217
732375663
-488763957
-29067534
560647387
546146545
636410338
927132810
401561021
937231062
516000492
-3658988
-190206527
639551123
624141577
719911943
-298495114
426716231
939358458
418356765
208443361
178270692
44169949
821968931
257114991
115268368
-287259330
-69030996
150068288
479935196
-468189320
-5434854
-234983815
45174859
-229665790
-470093734
730650817
-258105587
500285882
692601611
700824754
-52520066
219277565
64288199
332728252
-253116947
714382864
-348265291
769065730
550626038
258095842
-264743700
204609349
197939502
-146883194
121584213
67723112
666676749
-383544045
-214887524
239193137
287767902
-276421619
407599570
710383464
359914220
835422482
67810927
149959369
-211283950
585530619
283675045
136120400
277209889
-27691133
396605942
-204069627
602358458
-50948916
-242671194
656648977
557160115
666941296
593986174
365185793
-257871301
724269327
-20917
-438392775
-405040973
575818415
839813455
-90768414
-133793848
212195005
-143955621
258902684
488879737
245082632
751027595
887063498
-55774508
549193113
904737225
-76794998
180860432
608241143
490968488
221942495
995198104
-396888814
841803698
60244974
903802918
540122575
596261602
198875435
179861796
27744821
674439417
899318827
959213891
-111468334
-244266975
960764009
186451564
-169514730
296214477
804252464
458814538
771983445
-468273368
95808592
545082701
916798817
-5843067
-200350048
141251355
-336855210
110986336
-288868697
713173682
482535751
39427035
316474216
939990901
118538362
447226349
511971997
403285367
71194043
662517537
-337860451
-484989479
732751421
-109266518
528132743
392300181
252202758
-252868930
-472436450
-13293177
763493603
972446844
260780754
536606561
649847689
-147750650
499182536
-284631183
-365536706
492742338
713019687
565962158
481816090
523244534
280321439
168664106
-42077542
121921566
-463100091
700000066
75867
102148389
-1254945
-353295477
853217040
28173209
-174480358
974797704
899714765
585014562
-139145667
-38026894
888396678
-137853861
414033199
720804132
154468580
336729515
8386366
988440055
-255798883
983889321
481808915
725963510
600855234
985100012
21989949
-64952162
-424768825
-406970177
-303308547
881758601
-98045932
653233983
-364482672
283470838
697291470
566240283
-273461994
351396692
446721154
110662832
114052537
511766531
5567007
549011634
916351690
259577493
478044755
-13053815
946552063
370635016
724607699
312886208
-35120020
367167408
-24755537
382791753
-325084102
679973376
324216382
34880428
-88675376
-202527386
745136895
-304741018
871402375
-494724263
285518252
-191881043
364663785
-458973994
111255295
689652703
106755374
981917494
147094028
-250001508
202043915
-141603847
48478722
308855679
-469690162
985127394
405224035
191387570
-13559024
850022693
185224136
242294409
479988006
480528061
705167047
601934740
-434838058
617030507
651298484
994951042
766849061
-152460512
791949700
150909053
330825078
307604848
394812263
528478791
-400722174
531961770
-83789501
567237059
792472207
244754566
955367651
633179840
46314217
375275887
-149564904
-380640600
996180989
722113432
21541154
403672738
-122212142
-397928838
946757610
-302851674
904465529
-158595135
487581454
517751897
640737860
-244697849
673791539
492961609
569214838
564065826
611291655
-215793256
-314842739
765402082
25300178
58677109
370033035
-381251155
-97000744
283210724
-278614242
-41301097
656105389
-483639935
487669999
-420830291
678069046
706962765
-421612631
-138085756
971737788
-339892511
942964312
374035109
-129342816
-104170318
-127653230
-375343466
45455765
982184624
101286526
19596859
2796790
343511543
954468988
467068916
-301617682
-7438835
-358459643
-289247246
628061008
112423530
-420316556
-36590565
113537519
-450251421
507289767
-176825503
-99503722
970352802
385027941
-438317368
206906153
945871043
806142585
307312740
436626542
3140679
-333987218
-186545828
-398070045
275714592
598536543
966495083
-490510113
959565310
-312956194
-263718458
356783472
274907637
908477265
91318556
69213258
581932491
21606975
766377446
977216372
714395861
517389446
-114816180
-296304329
152143687
93108043
967336524
-362508097
-4952958
161556539
956427499
139633249
181700858
-206288557
68054363
87320980
-124742619
197677083
631198845
-110880872
294666258
-196651610
783595193
-156176159
162305582
-91056351
-209275402
927131137
-176318455
-115240759
623389411
-253234765
-489256349
671292363
-403539289
524140567
351193899
872059381
127676949
735954421
-154207523
-115503730
-323599163
-129213168
265906347
36688040
388383796
596913101
358696785
393963608
-126721944
560293404
383470439
326263975
683616469
958682900
144455476
412657178
-132864599
771377666
-100159559
-68328586
833126221
-154830399
468804013
-7145404
24322466
-25739
34065049
275704724
147263148
79893474
360728898
-241479721
-70673505
-410720232
-50519162
-277409746
-286226763
-106094513
-74147200
945059664
302542082
-380715795
467688539
508108049
694014619
423654930
-323257188
942418155
524906614
432949426
158891663
206975332
-152631669
-449117348
583649048
749276273
386609979
965821091
767734271
44915616
738842634
828276402
459424919
255801391
358006365
-242286604
-213296441
-218216740
880851188
778312400
856558554
86566212
240431390
326461473
-156194714
-356907904
-310551269
826837432
153956078
-302888751
-278061590
956029830
494612325
177763190
-477891201
-269067592
609628374
214381632
346684857
-197816811
917848319
878714668
171581454
791092278
-335799948
300361270
893212790
144853360
553971986
580692291
571314392
425964797
-264126450
884316764
484789556
148673228
841765777
-127030163
-134933313
694706
-354240196
627524015
44582850
876679288
-387359279
464117274
467481435
364204421
204619409
772040770
335064719
839074370
206123058
-149211240
301230186
638674778
600339934
11708272
85583657
979189588
-41038030
978531945
891132724
822595775
184035391
-205213137
-135082633
-370659317
411121995
469775160
990148971
581086998
571717390
-319277936
225476888
879206982
201594672
658096204
791003904
627245743
216374910
-21026735
711200180
140231618
226657366
244119777
68267050
-326510384
452379777
648355343
-85810458
-370447909
-85673100
45964835
-271200681
539313741
362610536
722007723
260461981
287676770
743708662
-34670947
526229313
946048853
-266197528
634126817
849296685
-13830937
-33457621
913588804
987527609
-409985667
624640007
-297607237
847043968
238424339
335836383
172944717
-293309546
-398670994
-328576657
-293360067
42523989
985606802
-370311326
759193107
829150769
-359718300
922680093
438956666
-99302962
70484952
387024323
-359747134
141843920
897786320
-186376383
437227719
-369493329
953995962
-468440368
-298228088
235707303
-365813195
448810959
965404941
71422257
269516493
427733060
34555584
513208496
857982446
42796654
-401833999
-498743590
-200248953
910914605
669532638
746108109
663637348
418984961
-418229245
706442110
787257412
968386314
678224565
451963335
915373026
347567109
693791581
-87989883
546351122
359345170
-102763608
-276405867
196335448
-433744274
267420705
29620652
207361627
-178059623
600711274
605944358
-169481689
-448199543
982548039
-262881370
-61518504
903086161
918878000
211050108
375709328
-276939622
746450128
225048182
970796658
86764129
895904357
-236647882
605308517
781390214
435904300
828214277
989973081
713164245
-77096901
367543613
164800299
328733298
784736239
95550463
263112498
576509368
-89948898
552949958
319660027
-407960196
384170819
770080045
549805157
332874757
-360112889
697752926
147555332
974089132
-16514177
874562413
-159623309
835629539
13531689
-355980757
280558268
239917065
847920969
687340222
-162445410
482440590
860898212
-134066137
-466654204
-439002362
39140001
756084016
971623572
866722050
-460209930
362350200
328738161
-174208172
853762606
-464041171
232903607
746125809
418219683
384425847
519421482
-484042984
-164823233
-20218033
-363213128
259975218
208709559
815052535
292286038
-448916297
94562532
981024967
-343319618
-351211209
-225561069
243589119
-322449969
138023704
749470550
592548259
124983469
-140098951
670960049
-414721117
54957983
-331647819
377222745
366627385
463803219
-482032761
924989143
332506444
115645318
852662331
479578545
784979905
-303136652
516527071
857019
-82768163
428610501
-69744355
697500496
832644974
337822944
320366825
705681724
672402272
718878801
727837414
966000719
214144523
27242107
835923309
9065045
509813048
-421109777
-36022638
-258186630
448135868
355388187
-147205171
-118927883
715600620
-178686141
-444348486
768234586
-35618720
972958363
-423354295
250110137
820798806
-73379022
297169188
276418448
-495981164
544853164
-43003931
225733223
40507700
30028482
-340847323
485523868
-350906667
-416937014
51313589
-182054256
495347732
723346904
-231902099
-321917315
361960823
82656539
328603555
479142960
733030285
815992571
862389755
218318752
-337876403
454008670
962443564
365633191
337186725
283270522
397548955
561302149
-424001950
313447123
142112546
-474667491
953187200
-42504191
17365656
661050956
290427114
255412005
-459958439
-40739234
614132222
-27345872
137116181
938491841
250232459
-268574868
333444505
-174280855
-190117995
-392777680
261494760
-91137750
-230362786
922603553
-127046825
152713441
964433453
352769954
979373346
790277720
313665427
-164242648
632574037
670308931
506661309
-350877394
-336002009
49992484
-366328015
746733
810119079
695740754
925430038
-467995653
500968417
639453703
-411577856
-70139002
-96519115
211389612
653715787
-498451665
-24204965
381719009
17175497
-4292369
563405394
28474792
421113104
-427897358
91461801
-158443869
782569017
21283128
567082893
-132422728
-389631314
657011169
763363729
431787720
186760817
446596118
-279697918
-224387993
112498898
314399652
133205863
988452264
708236993
-392334053
48074688
-361761539
902215616
48496771
282216177
358337623
-386995401
908963448
287800612
-49323730
222595944
-121036426
-457665407
-492525067
779934200
-487019401
-63225312
-456374147
-174885716
884723397
896503409
-277837462
921807022
695965036
971240960
693393440
28625504
-358491691
456714696
-106185870
184162850
715471475
-4991344
156086975
2100103
868442570
-240741783
912601528
358026113
-151346292
129930965
988607751
-410005756
676529662
239650241
567884784
884883934
965538741
51071839
112582735
5065347
601504208
-141102623
721369100
595497226
480993086
23032585
-113481121
-320185926
997512367
375606607
-18950407
-202199054
-218109898
245354983
916323472
-210069057
713755183
-160022392
569603376
143462577
212082956
-144397249
-245212110
204220310
-194546247
-146316210
741989322
-20828780
-325483444
712062340
-359089460
-248588093
-140239141
115966369
825482009
-153488587
645157278
48834199
722602411
-167307272
758027172
-190107697
825124704
779592944
335043546
564580130
917796529
793461920
616383049
149813356
439291895
914502352
349191018
293571499
-472570461
-28228766
305673262
358897488
-237854094
272650033
230461630
131268936
-204243960
412030101
-194503725
576863907
-458457693
280334665
30026516
524414687
766216410
-409963941
-147228229
555435088
-420170576
653503188
371750505
857958553
445132509
779881363
424249991
529849197
203047599
48330726
528381589
845369473
-103770325
39531394
742757833
-299681483
801784571
260479346
362967738
-50263567
268732611
