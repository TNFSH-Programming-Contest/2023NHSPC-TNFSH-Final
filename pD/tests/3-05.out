1240448887336
